src/uinput.rs

```diff
@@ -175,6 +178,47 @@ impl VirtualDevice {
         self.file.write_all(bytes)
     }
 
+    /// Get the syspath representing this uinput device.
+    ///
+    /// The syspath returned is the one of the input node itself (e.g.
+    /// `/sys/devices/virtual/input/input123`), not the syspath of the device node.
+    pub fn get_syspath(&mut self) -> io::Result<PathBuf> {
+        let mut bytes = vec![0u8; 256];
+        unsafe { sys::ui_get_sysname(self.file.as_raw_fd(), &mut bytes)? };
+
+        if let Some(end) = bytes.iter().position(|c| *c == 0) {
+            bytes.truncate(end);
+        }
+
+        let s = std::str::from_utf8(&bytes).expect("invalid sys path");
+        let mut path = PathBuf::from(SYSFS_PATH);
+        path.push(s);
+
+        Ok(path)
+    }
+
+    /// Get the syspaths of the corresponding device nodes in /dev/input.
+    #[cfg(not(feature = "tokio"))]
+    pub fn enumerate_dev_nodes(&mut self) -> io::Result<DevNodes> {
```

Comment:
Let's name these separately in case a crate user for some reason wants to use both, and also to reduce confusion. Adding `_blocking` to the fn name here is probably reasonable. That also removes the need to gate the blocking one on the absence of a feature.

Reply:
Thanks, I was thinking the same but wasn't sure whether to call the async version something with `_async` or something else.